### ISSUE TITLE
swatch-renderer-mixin.js is broken PDP

### ISSUE THREAD

**Issue by fernandofauth** [fastly/fastly-magento2#455]
swatch-renderer-mixin.js is broken PDP when config is undefined

`
if (_.isUndefined(config.jsonConfig.images))
     return create(config, element);
`

**Comment by jordanvector** [comment]
This breaks all config products with a js type error on the page that looks like this

```
swatch-renderer-mixin.js:11 Uncaught TypeError: Cannot read property 'jsonConfig' of undefined
    at Object.<anonymous> (swatch-renderer-mixin.js:11)
    at new <anonymous> (wrapper.js:78)
    at Function.$.widget (widget.js:75)
    at swatch-renderer.js:1
    at mixins.js:105
    at Array.forEach (<anonymous>)
    at applyMixins (mixins.js:104)
    at mixins.js:129
    at Object.execCb (require.js:1650)
    at Module.check (require.js:866)
```

**Comment by jorgb90** [comment]
Serious problem..

**Comment by jordanvector** [comment]
@jorgb90 We downgraded for now to fix this issue

**Comment by vvuksan** [comment]
Thanks for the report. We have pulled the 1.2.157 version while we address the issue.

**Comment by Lukas713** [comment]
Hi @jordanvector, @fernandofauth  , I've added an additional validation to validate just config object. [PR](https://github.com/fastly/fastly-magento2/pull/456) . 

But before merge, can you give me example of configurable product that has now config object at pdp. 
Thanks

**Comment by jorgb90** [comment]
@Lukas713 We literally had it for every configurable product on our website.

**Comment by Lukas713** [comment]
@jorgb90 is there a chance that you send me a link to a staging server. Or gie me a magento store version.

**Comment by Lukas713** [comment]
@jorgb90 @jordanvector @fernandofauth 

the mixin is tested with Magento 2.3.0 and Magento 2.4.2 with various configurable products. Can you give me scenario where config object is empty but images are rendered.

**Comment by pmarki** [comment]
We have the same issue with latest Magento 2.3.6-p1.

**Comment by Lukas713** [comment]
Hi @pmarki 

I've just tested with magento 2.3.6 using luma theme. Can you give me more info to reproduce the issue.

Thanks

**Comment by Lukas713** [comment]
Hi @pmarki , @jorgb90 , @jordanvector, @fernandofauth 

I've made [additional commit](https://github.com/fastly/fastly-magento2/pull/456/files) that also validates config object at swatch-renderer-mixin.js

I've tested it with magento 2.3.6, 2.4.0 and 2.4.2 with magento luma theme and sample data. It seems that it works as expected.

Can you give any feedback on this. Thanks

**Comment by DanilSA** [comment]
Please, do not use **wrapper.wrap** method to extend the jQuery Widget!

https://devdocs.magento.com/guides/v2.4/javascript-dev-guide/javascript/js_mixins.html#extend-jquery-widget

**Comment by jorgb90** [comment]
@Lukas713 We are running 2.4.2-p1. I can give you access to a dev environment, if you still want to. We are running configurables with visual and text swatches, not dropdowns.

**Comment by Lukas713** [comment]
@jorgb90 you can check [here](https://favicodestaging.global.ssl.fastly.net/breathe-easy-tank.html). Size and color are text value and swatch value. `this.options` are never undefined.

However, I've added additional config validation to the [pull request](https://github.com/fastly/fastly-magento2/pull/457/files).

**Comment by dpotkoc** [comment]
Hello @jorgb90  it will be helpful if you could provide us access to dev server, also please send us exactly steps how to reproduce this issue.

Thanks in advance
Domagoj

**Comment by dpotkoc** [comment]
@pmarki @jordanvector @fernandofauth Please, could you send me exactly steps how to reproduce this issue.

Thanks in advance
Domagoj

**Comment by jorgb90** [comment]
> @jorgb90 you can check [here](https://favicodestaging.global.ssl.fastly.net/breathe-easy-tank.html). Size and color are text value and swatch value. `this.options` property is not undefined.
> 
> However, I've added additional config validation to the [pull request](https://github.com/fastly/fastly-magento2/pull/457/files).

I dont see any visual or text swatches on your link? I only see dropdowns. You should set the attributes accordingly.

**Comment by Lukas713** [comment]
Hi @jorgb90 ,

size and color are changed in visual swatch and text swatch. `this.options` or `this.options.jsonConfig` are still not `undefined`. 

You can check [here](https://favicodestaging.global.ssl.fastly.net/breathe-easy-tank.html?id=10001) .

**Comment by dpotkoc** [comment]
Hi @jorgb90 @pmarki  @jordanvector @fernandofauth any chance to get steps how to reproduce this issue.

I will be really appreciate. 

I am testing on 2.4.2 and I can't get this error. In next days we will make update on  swatch-renderer-mixin.js and make it on properly Magento way.

Thanks
Domagoj

**Comment by jorgb90** [comment]
> Hi @jorgb90 ,
> 
> size and color are changed in visual swatch and text swatch. `this.options` or `this.options.jsonConfig` properties are still not `undefined`, neither is `config` object `undefined` (if wrapper is used).
> 
> You can check [here](https://favicodestaging.global.ssl.fastly.net/breathe-easy-tank.html?id=10001) .

You mean its working now with your commits or also without? I havent tested your commits as of yet.

**Comment by jorgb90** [comment]
> Hi @jorgb90 @pmarki @jordanvector @fernandofauth any chance to get steps how to reproduce this issue.
> 
> I will be really appreciate.
> 
> I am testing on 2.4.2 and I can't get this error. In next days we will make update on swatch-renderer-mixin.js and make it on properly Magento way.
> 
> Thanks
> Domagoj

I am not doing anything special just have configurable products in Magento which uses swatches. As I said above I havent tested the commits created by @Lukas713 as of yet, so I guess that will fix the issue?

I havent connected anything to the Fastly service at the moment, but I can try it out today on our dev. Hopefully @pmarki @jordanvector @fernandofauth can also provide some answers.

**Comment by jorgb90** [comment]
@dpotkoc @Lukas713 Sorry it took a while but with your commits everything is working as expected.

**Comment by vvuksan** [comment]
Fixed with #457. Will be part of 1.2.158.

**Comment by jordanvector** [comment]
I can confirm this is resolved in latest version  1.2.158
